### PR TITLE
Implemented functionality to fetch order with the help of solr query and show those orders in FindOrder.vue page ( #1x634dq ).

### PR DESCRIPTION
**Pull request by meet-aniket**

### CODE REVIEW

File: src/views/FindOrder.vue [hunk 1]
```diff
@@ -104,61 +104,80 @@
...
                   <ion-icon :icon="ribbon" />
-                  <ion-label>Customer Loyalty Status</ion-label>
+                  <ion-label> {{ $filters.getCustomerLoyalty(order.doclist.docs[0].orderNotes, 'cusotmerLoyaltyOptions') }} </ion-label>
                 </ion-chip>
               </div>
 
               <div class="metadata">
-                <ion-note>Ordered on 7 Jan 2021</ion-note>
-                <ion-badge>Approved</ion-badge>
+                <ion-note> {{ $t("Ordered on") }} {{ $filters.formatUtcDate(order.doclist.docs[0].orderDate, 'YYYY-MM-DDTHH:mm:ssZ') }} </ion-note>
```
Comment (dt2patel) [inline src/views/FindOrder.vue]: @ymaheshwari1 what is the format that we decided on for showing times using moment library? we should document that somewhere and make it easily reference-able for everyone.

File: src/views/FindOrder.vue [hunk 2]
```diff
@@ -104,61 +104,80 @@
...
+                <ion-badge> {{ order.doclist.docs[0].orderStatusId }} </ion-badge>
               </div>
             </section>
 
-            <section class="section-grid" @click="() => router.push('/order')">
-              <div v-for="item in 7" :key="item">
+            <section class="section-grid" @click="() => router.push(`/order/${order.doclist.docs[0].orderId}`)">
```
Comment (dt2patel) [inline src/views/FindOrder.vue]: need to move this click action up to wrapper div of order at line 107. Should have been done during static UI work but looks like we missed it.

File: src/views/FindOrder.vue [hunk 3]
```diff
@@ -104,61 +104,80 @@
...
+              <div v-for="(item, index) in order.doclist.docs" :key="index" :item="item">
```
Comment (dt2patel) [inline src/views/FindOrder.vue]: Can this logic not be applied directly to the card?

File: src/views/FindOrder.vue [hunk 4]
```diff
@@ -104,61 +104,80 @@
...
+                      <Image :src="getProduct(item.productId).mainImageUrl" />
                     </ion-thumbnail>
                     <ion-label>
-                      <p>Brand</p>
-                      Virtual Name
-                      <p>{{ $t("Color") }} : color</p>
-                      <p>{{ $t("Size") }}: size</p>
+                      <p> {{ getProduct(item.productId).brandName ? getProduct(item.productId).brandName : '-' }} </p>
+                      {{ item.virtualProductName }}
+                      <p> {{ $t("Color") }} : {{ $filters.getFeature(getProduct(item.productId).featureHierarchy, '1/COLOR/') }} </p>
```
Comment (dt2patel) [inline src/views/FindOrder.vue]: Add a todo comment here to make this dynamic, where feature key and value will both be dynamically pulled from source. Different retailers will have different product feature types so we cannot hardcode this.

File: src/views/FindOrder.vue [hunk 5]
```diff
@@ -104,61 +104,80 @@
...
-                    <ion-label> {{ $t("PO arrival date") }} </ion-label>
-                    <p slot="end">14 Jan 2021</p>
-                  </ion-item>
-                  <ion-item>
-                    <ion-label> {{ $t("Last brokered") }} </ion-label>
-                    <p slot="end">California Warehouse</p>
+                    <ion-badge color="primary" slot="end"> {{ item.orderItemStatusId }} </ion-badge>
                   </ion-item>
+                  <!-- TODO: Need to handle this property -->
+                  <div v-if="item.preOrderStatus || item.backOrderStatus || item.unFillable">
```
Comment (dt2patel) [inline src/views/FindOrder.vue]: We can remove unfillable check from here. this is specifically for pre order or backorder. I like the thought though.

File: src/views/FindOrder.vue [hunk 6]
```diff
@@ -104,61 +104,80 @@
...
+                    <ion-item>
+                      <ion-label> {{ $t("Promise date") }} </ion-label>
+                      <p slot="end"> {{ item.promisedDatetime ? $filters.formatUtcDate(item.promisedDatetime, 'YYYY-MM-DDTHH:mm:ssZ') : '-'  }} </p>
```
Comment (dt2patel) [inline src/views/FindOrder.vue]: Again with the dates, please ask @ymaheshwari1 what the standard we decided on was.

File: src/views/FindOrder.vue [hunk 7]
```diff
@@ -104,61 +104,80 @@
...
+                    </ion-item>
+                  </div>
+                  <div v-else>
+                    <ion-item>
+                      <ion-label> {{ $t("Shipping method") }} </ion-label>
+                      <p slot="end"> {{ item.shipmentMethodTypeId }} </p>
+                    </ion-item>
+                    <ion-item>
+                      <ion-label> {{ $t("Shipping from") }} </ion-label>
+                      <p slot="end"> {{ $filters.getShippingFrom(item.orderRoles, '1/Ship-From Vendor/') ? $filters.getShippingFrom(item.orderRoles, '1/Ship-From Vendor/') : "-" }} </p>
```
Comment (dt2patel) [inline src/views/FindOrder.vue]: we should probably be using orderFacilityName or [facilityName](https://git.hotwax.co/commerce/oms/blob/develop/applications/hwmapps/solrconfig/solrschema.xml#L198)

@dixitdeepak @patelanil what do you think?

File: src/views/FindOrder.vue [hunk 8]
```diff
@@ -104,61 +104,80 @@
...
+                    </ion-item>
+                    <ion-item>
+                      <ion-label> {{ $t("Location inventory") }} </ion-label>
+                      <p slot="end"> {{item.uniqueOrderItemsCount}} </p>
```
Comment (dt2patel) [inline src/views/FindOrder.vue]: Need to get inventory for given item at the location it is shipping from. Will require getting productID and checking inventory at the facilityID of the order item. @adityasharma7 what do you think?

**Review by dt2patel — CHANGES_REQUESTED** [review]
Need some work on the data we are showing and how we get it.

File: src/views/FindOrder.vue (hunk 6 above)
Comment (ymaheshwari1) [reply src/views/FindOrder.vue]: Sir, the format for date is `Jan 26th 2022` (MMM DD YYYY).

File: src/views/FindOrder.vue (hunk 2 above)
Comment (ymaheshwari1) [reply src/views/FindOrder.vue]: Moved the routing logic to div containing whole order info.

File: src/views/FindOrder.vue (hunk 3 above)
Comment (ymaheshwari1) [reply src/views/FindOrder.vue]: Sir, Added the looping logic to the ion-card.

File: src/views/FindOrder.vue (hunk 5 above)
Comment (ymaheshwari1) [reply src/views/FindOrder.vue]: Sir, removed the check for unfillable status.

File: src/views/FindOrder.vue (hunk 6 above)
Comment (ymaheshwari1) [reply src/views/FindOrder.vue]: Sir, added the date format `D MMM YYYY` as per the figma.

File: src/views/FindOrder.vue (hunk 8 above)
Comment (ymaheshwari1) [reply src/views/FindOrder.vue]: Sir, added logic to get the product inventory at a specific facility.

File: src/views/FindOrder.vue (hunk 4 above)
Comment (ymaheshwari1) [reply src/views/FindOrder.vue]: Added a todo and will handle this in another task.

File: src/store/RootState.ts [hunk 9]
```diff
@@ -1,4 +1,5 @@
 export default interface RootState {
     user: any;
     product: any;
+    orders: any;
```
Comment (adityasharma7) [inline src/store/RootState.ts]:
```suggestion
    order: any;
```

File: src/views/FindOrder.vue [hunk 10]
```diff
@@ -235,19 +265,81 @@ export default {
...
+    }
+  },
+  mounted() {
+    this.getOrders();
+  },
   setup() {
     const router = useRouter();
+    const store = useStore();
+    const queryString = ref();
+    const orderStatus = reactive(JSON.parse(process.env.VUE_APP_ORDER_STATUS))
```
Comment (adityasharma7) [inline src/views/FindOrder.vue]: I think we could skip making it reactive

File: src/views/FindOrder.vue [hunk 11]
```diff
@@ -104,75 +104,94 @@
...
 
-          <div @click="() => router.push('/order')">
+          <div v-for="(order, index) in orders" :key="index" :order="order" @click="() => router.push(`/order/${order.doclist.docs[0].orderId}`)">
             <section class="section-header">
               <div class="primary-info">
                 <ion-item lines="none">
                   <ion-label>
-                    Order ID
-                    <p>Customer Name</p>
+                    {{ order.doclist.docs[0].orderId }}
```
Comment (adityasharma7) [inline src/views/FindOrder.vue]: I think orderId should be available directly on order object and we should perform such extraction in actions. Same applies to other fields here

**Review by adityasharma7 — CHANGES_REQUESTED** [review]

File: src/views/FindOrder.vue (hunk 11 above)
Comment (ymaheshwari1) [reply src/views/FindOrder.vue]: Extracted the commonly used fields in action and then used them directly.

File: src/views/FindOrder.vue (hunk 10 above)
Comment (ymaheshwari1) [reply src/views/FindOrder.vue]: Sir, removed making the status object reactive.

File: src/store/RootState.ts (hunk 9 above)
Comment (ymaheshwari1) [reply src/store/RootState.ts]: Sir, Changed the module name in rootstate.

**Review by adityasharma7 — APPROVED** [review]

**Pull request merged**